### PR TITLE
Update renovate config to match template

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Update the renovate config to better match the template. 

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: .github/renovate.json [hunk 1]
```diff
@@ -2,26 +2,25 @@
...
-      "matchUpdateTypes": ["minor", "patch"],
-      "groupName": "npm"
+      "groupName": "{{manager}}",
+      "matchPackagePatterns": ["*"],
+      "matchUpdateTypes": ["minor", "patch"]
     },
     {
       "packageNames": ["typescript"],
-      "updateTypes": "patch"
+      "matchUpdateTypes": "patch"
```
Comment (withinfocus) [inline .github/renovate.json:23]: ❓ No minors wanted here?

Comment (Hinton) [reply .github/renovate.json:23]: No, we need to align with the minor versions Angular supports.

Comment (MGibson1) [reply .github/renovate.json:23]: For context on the TS versioning: https://github.com/microsoft/TypeScript/issues/14116#issuecomment-280592571

**Review by MGibson1 — APPROVED** [review]
Looks like a good conversion to me, @withinfocus, want to review again or move forward?

**Review by withinfocus — APPROVED** [review]
Makes sense -- thanks.

**Pull request merged**